transformer_engine/common/transpose/cast_transpose_fusion.cu

```diff
@@ -635,71 +755,121 @@ void cast_transpose_fused(const Tensor &input,
-                static_assert(shared_size_transpose >= shared_size_dbias);
+                size_t VecComputeTypeSize;
+                switch (nvec_in) {
+                    case 1: VecComputeTypeSize = sizeof(Vec<ComputeType, 1>); break;
+                    case 2: VecComputeTypeSize = sizeof(Vec<ComputeType, 2>); break;
+                    case 4: VecComputeTypeSize = sizeof(Vec<ComputeType, 4>); break;
+                    case 8: VecComputeTypeSize = sizeof(Vec<ComputeType, 8>); break;
+                }
+                const size_t shared_size_dbias = cast_transpose_num_threads * VecComputeTypeSize;
+                if (shared_size_transpose < shared_size_dbias) {
+                    shared_size_transpose = shared_size_dbias;
+                    // NVTE_CHECK(shared_size_transpose >= shared_size_dbias,
+                    //         "Too large size of the dbias shared memory required");
+                }
             }
 
-            using Param = CTDBiasDGeluParam<InputType, InputType2, OutputType, ComputeType>;
             Param param;
-            param.input = reinterpret_cast<const InputType *>(input.data.dptr);
-            param.output_c = reinterpret_cast<OutputType *>(cast_output->data.dptr);
-            param.output_t = reinterpret_cast<OutputType *>(transposed_output->data.dptr);
-            param.scale_ptr = reinterpret_cast<const ComputeType *>(cast_output->scale.dptr);
-            param.amax = reinterpret_cast<ComputeType *>(cast_output->amax.dptr);
+            param.input      = reinterpret_cast<const InputType *>(input.data.dptr);
+            param.output_c   = reinterpret_cast<OutputType *>(cast_output->data.dptr);
+            param.output_t   = reinterpret_cast<OutputType *>(transposed_output->data.dptr);
+            param.scale_ptr  = reinterpret_cast<const ComputeType *>(transposed_output->scale.dptr);
+            param.amax       = reinterpret_cast<ComputeType *>(transposed_output->amax.dptr);
+            param.scale_inv  = reinterpret_cast<ComputeType *>(cast_output->scale_inv.dptr);
             if constexpr (IS_DBIAS) {
                 param.workspace = reinterpret_cast<ComputeType *>(workspace->data.dptr);
             }
             if constexpr (IS_DACT) {
                 param.act_input = reinterpret_cast<const InputType2 *>(act_input.data.dptr);
             }
 
-            if (full_tile) {
-                cudaFuncSetAttribute(
+            // Runtime-compiled tuned kernel
+            if (jit_compiled) {
+                constexpr const char *itype_name  = TypeInfo<InputType>::name;
+                constexpr const char *itype2_name = TypeInfo<InputType2>::name;
+                constexpr const char *otype_name  = TypeInfo<OutputType>::name;
+
+                // Compile NVRTC kernel if needed and launch
+                auto& rtc_manager = rtc::KernelManager::instance();
+                const std::string kernel_label = concat_strings("cast_transpose"
```

Comment:
To make sure this key is distinct from https://github.com/NVIDIA/TransformerEngine/blob/89d44e2759f27c690f819e6d48a695081b900b62/transformer_engine/common/transpose/cast_transpose.cu#L317
```suggestion
                const std::string kernel_label = concat_strings("cast_transpose_fused"
```